core/src/apps/management/reset_device/layout.py

```diff
@@ -208,113 +278,70 @@
-
-        paginated.on_change = export_displayed_words
-        export_displayed_words()
-
-    await hold_to_confirm(ctx, paginated, ButtonRequestType.ResetDevice)
-
-
-def _get_mnemonic_page(words: list):
-    text = Text("Recovery seed", ui.ICON_RESET)
-    for index, word in words:
-        text.mono("%2d. %s" % (index + 1, word))
-    return text
-
-
 # SLIP39
 # ===
 
-# TODO: yellow cancel style?
-# TODO: loading animation style?
-# TODO: smaller font or tighter rows to fit more text in
-# TODO: icons in checklist
-
-# SLIP 39 simple
-
 
-async def slip39_show_checklist_set_shares(ctx):
+async def slip39_show_checklist(ctx, step: int, backup_type: BackupType) -> None:
     checklist = Checklist("Backup checklist", ui.ICON_RESET)
-    checklist.add("Set number of shares")
-    checklist.add("Set threshold")
-    checklist.add(("Write down and check", "all recovery shares"))
-    checklist.select(0)
-    return await confirm(
-        ctx, checklist, ButtonRequestType.ResetDevice, cancel=None, confirm="Continue"
-    )
-
-
-async def slip39_show_checklist_set_threshold(ctx, num_of_shares):
-    checklist = Checklist("Backup checklist", ui.ICON_RESET)
-    checklist.add("Set number of shares")
-    checklist.add("Set threshold")
-    checklist.add(("Write down and check", "all recovery shares"))
-    checklist.select(1)
-    return await confirm(
-        ctx, checklist, ButtonRequestType.ResetDevice, cancel=None, confirm="Continue"
-    )
-
+    if backup_type is BackupType.Slip39_Basic:
+        checklist.add("Set number of shares")
+        checklist.add("Set threshold")
+        checklist.add(("Write down and check", "all recovery shares"))
+    elif backup_type is BackupType.Slip39_Advanced:
+        checklist.add("Set number of groups")
+        checklist.add("Set group threshold")
+        checklist.add(("Set size and threshold", "for each group"))
+    checklist.select(step)
 
-async def slip39_show_checklist_show_shares(ctx, num_of_shares, threshold):
-    checklist = Checklist("Backup checklist", ui.ICON_RESET)
-    checklist.add("Set number of shares")
-    checklist.add("Set threshold")
-    checklist.add(("Write down and check", "all recovery shares"))
-    checklist.select(2)
     return await confirm(
         ctx, checklist, ButtonRequestType.ResetDevice, cancel=None, confirm="Continue"
     )
 
 
-# SLIP 39 group
-
-
-async def slip39_group_show_checklist_set_groups(ctx):
-    checklist = Checklist("Backup checklist", ui.ICON_RESET)
-    checklist.add("Set number of groups")
-    checklist.add("Set group threshold")
-    checklist.add(("Set size and threshold", "for each group"))
-    checklist.select(0)
-    return await confirm(
-        ctx, checklist, ButtonRequestType.ResetDevice, cancel=None, confirm="Continue"
-    )
-
+async def slip39_prompt_threshold(ctx, num_of_shares, group_id=None):
+    count = num_of_shares // 2 + 1
+    min_count = min(2, num_of_shares)
```

Comment:
The `min` is weird. It is always 2, unless `num_of_shares` is 1? Is this the thing with the 1-of-1 share? If so, an explanatory comment would help.

Reply:
added a comment to clarify